sdks/python/apache_beam/dataframe/frames_test.py

```diff
@@ -1245,6 +1245,37 @@ def test_idxmax(self):
     self._run_test(lambda s2: s2.idxmax(), s2)
     self._run_test(lambda s2: s2.idxmax(skipna=False), s2)
 
+  def test_pipe(self):
+    def df_times(df, column, times):
+      df[column] = df[column] * times
+      return df
+
+    def s_times(s, times):
+      return s * times
+
+    df = pd.DataFrame({'A': [1, 2, 3], 'B': [4, 5, 6]}, index=[0, 1, 2])
+    s = pd.Series([1, 2, 3, 4, 5], index=[0, 1, 2, 3, 4])
+    # s2 = pd.Series([6, 7], index=[5, 6])
+
+    func_1 = df_times
+    func_2 = frames.DeferredDataFrame.sum
+    func_3 = frames.DeferredSeries.sum
+    func_4 = s_times
+    # func_5 = frames.DeferredSeries.append
+
+    self._run_inplace_test(lambda df: df.pipe(func_1, 'A', 2), df)
+    self._run_test(lambda df: df.pipe(func_2), df)
+    # type assert fails when axis=1
+    # self._run_test(lambda df: df.pipe(func_2, axis=1), df)
+    self._run_inplace_test(lambda df: df.pipe(func_1, 'A', 2).pipe(func_2), df)
+    self._run_test(lambda df: df.pipe(func_2).pipe(func_3), df)
+
+    self._run_test(lambda s: s.pipe(func_4, 2), s)
+    self._run_test(lambda s: s.pipe(func_3), s)
+    self._run_test(lambda s: s.pipe(func_4, 2).pipe(func_3), s)
+    # Can't append non-deferred series
+    # self._run_test(lambda s: s.pipe(func_5, s2).pipe(func_4, 2), s)
```

Comment:
Does `s2` need to be wrapped as a DeferredSeries to make it work?

Reply:
You can pass multiple frames to `_run_test` and they will all be converted to Deferred counterparts and passed to the lambda (for the distributed run). I think this should work:

```suggestion
    self._run_test(lambda s, s2: s.pipe(func_5, s2).pipe(func_4, 2), s, s2)
```